source/extensions/network/dns_resolver/getaddrinfo/getaddrinfo.cc

```diff
@@ -21,23 +21,41 @@ class AddrInfoWrapper : NonCopyable {
 
 } // namespace
 
+GetAddrInfoDnsResolver::GetAddrInfoDnsResolver(
+    const envoy::extensions::network::dns_resolver::getaddrinfo::v3::GetAddrInfoDnsResolverConfig&
+        config,
+    Event::Dispatcher& dispatcher, Api::Api& api)
+    : config_(config), dispatcher_(dispatcher) {
+  uint32_t num_threads =
+      config_.has_num_resolver_threads() ? 1 : config_.num_resolver_threads().value();
```

Comment:
This should be the opposite, `config_.has_num_resolver_threads() ? config_num_resolver_threads().value() : 1;`